### PR TITLE
Amstrad PCW driver step I

### PR DESCRIPTION
**Pull request by JoakimLarsson**
I have sorted out differences between the different pcw models. 

Will take a look at why the second floppy doesn't work and go through the softlist later, will come in separate PR:s later

### CODE REVIEW

File: src/mame/drivers/pcw.cpp [hunk 1]
```diff
@@ -246,7 +246,7 @@ void pcw_state::pcw_update_read_memory_block(int block, int bank)
 	address_space &space = m_maincpu->space(AS_PROGRAM);
 	char block_name[10];
 
-	sprintf(block_name,"bank%d",block+1);
+	snprintf(block_name, sizeof(block_name), "bank%d", block + 1);
```
Comment (cuavas) [inline src/mame/drivers/pcw.cpp]: Eugh, isn't this a perfect case for `required_memory_bank_array` so you get benefits in validation, performance, and expressiveness?

File: src/mame/drivers/pcw.cpp [hunk 2]
```diff
@@ -1020,6 +1021,7 @@ void pcw_state::machine_reset()
 	m_printer_headpos = 0x00; // bring printer head to left margin
 	m_printer_shift = 0;
 	m_printer_shift_output = 0;
+	machine().render().first_target()->set_view(0);
```
Comment (cuavas) [inline src/mame/drivers/pcw.cpp]: This breaks expectations.  MAME saves/restores selected view automatically, and you've broken that.  It also won't play nicely with external artwork either (the first view will come from the external artwork, so you can't make assumptions about it), it probably breaks the `-view0` etc. command-line options, and if the user configures the system to have an additional screen (e.g. with a terminal) it won't show by default.

Comment (JoakimLarsson) [reply src/mame/drivers/pcw.cpp]: ok, I just want to get rid of the printer screen on startup, it ain't useful until you print something and de-centers the terminal view

File: src/mame/drivers/pcw.cpp (hunk 1 above)
Comment (JoakimLarsson) [reply src/mame/drivers/pcw.cpp]: You already did, didn't you :)

**Pull request merged**